x-pack/plugins/security_solution/common/types/header_actions/index.ts

```diff
@@ -103,11 +103,16 @@ export interface ActionProps {
   setEventsDeleted: SetEventsDeleted;
   setEventsLoading: SetEventsLoading;
   showCheckboxes: boolean;
+  /**
+   * This prop is used to determine if the notes button should be displayed
+   * as the part of Row Actions
+   * */
   showNotes?: boolean;
   tabType?: string;
   timelineId: string;
   toggleShowNotes?: () => void;
   width?: number;
+  disablePinAction?: boolean;
```

Comment:
any reason for not calling this `showPinAction` instead? we're not really disabling the button right? we're actually hiding it?

Reply:
No particular reason but recently michael added `disabledExpandAction` event so I just wanted consistence. I want to rename `showNotes` to `disableNotesAction` event as well but I am trying to limit the scope of this PR.

Reply:
ok, I'm ok with it... I still feel like `show` is better than `disable` as it better represents what the property is actually doing... But I guess that can be cleaned up in a follow up PR!

Reply:
Created an issue : https://github.com/elastic/security-team/issues/9866

Reply:
what if we had consistency with `show` instead of `disable`?

We already have `showNotes` so if you add `showPinEvent` instead of `disablePinEvent` then we only have to change `disableExpandAction` into `showExpand` in the follow up PR...

Anyway, I'll stop as it doesn't matter than much 😆

Reply:
>Anyway, I'll stop as it doesn't matter than much 😆

I agree and let's tackle that in other issue.

Reply:
there are cases where the pin button is shown but disabled, when a note is associated with an event in a timeline, that event is considered pinned, and so a prop with this name is used to disable the onClick and gray out the icon.

Reply:
@kqualters-elastic . Thanks kevin. Let me check it out.

Reply:
Hey @kqualters-elastic, @PhilippeOberti  .. what should be ideal behaviour when note is attached to an event? Do we consider it pinned or not?